designer/server/src/test/scala/pl/touk/nussknacker/ui/api/MigrationApiEndpointsBusinessSpec.scala

```diff
@@ -0,0 +1,187 @@
+    with WithRichConfigRestAssuredUsersExtensions
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers
+    with RestAssuredVerboseLogging {
+
+  private val sourceEnvironmentId = "DEV"
+
+  private val exampleProcessName = ProcessName("test2")
+  private val illegalProcessName = ProcessName("#test")
+
+  private val exampleScenario =
+    ScenarioBuilder
+      .withCustomMetaData(exampleProcessName.value, Map("environment" -> "test"))
+      .source("source", "csv-source-lite")
+      .emptySink("sink", "dead-end-lite")
+
+  private val validFragment =
+    ScenarioBuilder.fragmentWithInputNodeId("source", "csv-source-lite").emptySink("sink", "dead-end-lite")
+
+  private val exampleGraph = CanonicalProcessConverter.toScenarioGraph(exampleScenario)
+
+  private val successValidationResult = ValidationResult.success
+
+  private val errorValidationResult =
+    ValidationResult.errors(
+      Map("n1" -> List(NodeValidationError("bad", "message", "", None, NodeValidationErrorType.SaveAllowed))),
+      List(),
+      List()
+    )
+
+  private def prepareRequestJsonData(
+      scenarioName: String,
+      validationResult: ValidationResult,
+      scenarioGraph: ScenarioGraph,
+      isFragment: Boolean
+  ): String =
+    s"""
+       |{
+       |  "sourceEnvironmentId": "$sourceEnvironmentId",
+       |  "processingMode": "Unbounded-Stream",
+       |  "engineSetupName": "Flink",
+       |  "scenarioWithDetailsForMigrations": {
+       |    "name": "${scenarioName}",
+       |    "isArchived": false,
+       |    "isFragment": $isFragment,
+       |    "processingType": "streaming1",
+       |    "processCategory": "Category1",
+       |    "scenarioGraph": ${scenarioGraph.asJson.noSpaces},
+       |    "validationResult": ${validationResult.asJson.noSpaces},
+       |    "history": null,
+       |    "modelVersion": null
+       |  }
+       |}
+       |""".stripMargin
+
+  private val validRequestData: String =
+    prepareRequestJsonData(exampleProcessName.value, successValidationResult, exampleGraph, false)
+
+  private val invalidRequestData: String =
+    prepareRequestJsonData(exampleProcessName.value, errorValidationResult, exampleGraph, false)
+
+  private val requestDataWithInvalidScenarioName: String =
+    prepareRequestJsonData(illegalProcessName.value, successValidationResult, exampleGraph, false)
+
+  private val validRequestDataForFragment: String =
+    prepareRequestJsonData(exampleProcessName.value, successValidationResult, exampleGraph, true)
+
+  "The endpoint for scenario migration between environments should" - {
+    "migrate scenario and add update comment" in {
+      given()
+        .applicationState(
+          createSavedScenario(exampleScenario, Category1)
+        )
+        .when()
+        .basicAuthAllPermUser()
+        .jsonBody(validRequestData)
+        .post(s"$nuDesignerHttpAddress/api/migrate")
+        .Then()
+        .statusCode(200)
+
+      given()
```

Comment:
this is the assertion right? 
Please, take a look at the `ScenarioActivityApiHttpServiceBusinessSpec` and the `verifyCommentExists` assertion. Should we extract it to some helper? (eg. `NuRestAssureExtensions`?)